### DISCUSSION TITLE
Accessing PSK Identity + Key used in DTLS Connection

### DISCUSSION THREAD

**Discussion by motters** [webrtc-rs/webrtc#292] [Q&A]
Hi All,

I've been playing with your DTLS crate for use with a COAP server. Everything is working great, you've done an amazing job!

One limitation I'm encountering is once a connection has been [‘accepted’](https://github.com/webrtc-rs/webrtc/blob/master/dtls/examples/listen/psk/listen_psk.rs#L79), you can't get access to the authentication details used by the client. 

This is because the DTLSConn is [downcast](https://github.com/webrtc-rs/webrtc/blob/master/dtls/src/listener.rs#L74) to the [Conn trait](https://github.com/webrtc-rs/webrtc/blob/master/util/src/conn/mod.rs#L27).

For example, if a client connects using a PSK identity + key, is there a work around to get the identity + key used for the ‘accepted’ connection?

This is required when you want to match a connection to a registered client for protocols such as LwM2M.

Thanks in Advance!

**Comment by k0nserv** [comment]
Hey, 

Thanks for trying out the DTLS crate. While the primary use case for the crate is a as part of the webrtc-rs project, we welcome other usage too. 

Personally I'm not super familiar with the DTLS crate(it was written by @rainliu), so I'm afraid I don't have an answer to your question. If you'd like to contribute a change to unblock your use case we will accept it, as long as it doesn't compromise the use webrtc-rs use case.
